### DISCUSSION TITLE
Use ory Hydra as sso

### DISCUSSION THREAD

**Discussion by elhamenayati** [ory/hydra#3534] [Q&A] [answered]
In my application, I use minio as my file manager. When user logs in my application, it has to be logged in both minio. How can I use ory hydra in this case? Do I have to use both ory Hydra and Kratos?

**Comment by vinckr** [comment, marked as answer]
Hello @elhamenayati 

I don't know enough about Minio to give you a confident answer here.

Ory Hydra is an OAuth 2.0 and OpenID Connect provider, and Ory Kratos is an identity management server. They can be used separately or together depending on your specific needs for authentication and user management.

If you want to federate an identity across different applications or domains, meaning you can sign in to both apps with the same credentials, then OAuth2 can be a good solution. 
[Ory Network](https://www.ory.sh/network/) also offers a multi-domain feature that can be used for this, so you dont have to code your own.

For more detailed information on how to use Ory Hydra and Ory Kratos, you can refer to the official Ory documentation [here](https://www.ory.sh/docs/ecosystem/projects) and [here](https://www.ory.sh/docs/kratos/ory-kratos-intro).

**Reply by baominwang** [reply]
If we exclusively plan to utilize Hydra, is it feasible to enable SSO? Or do we need to develop custom code in our private identity management server? Alternatively, should we integrate Ory Kratos for SSO support?
Is there any related guide?

**Reply by vinckr** [reply]
Hey @baominwang 

You can also just use Ory Hydra. Again it depends a lot on what you want to achieve, what SSO means to you - can you elaborate a bit more what you want to achieve or what you understand as SSO? 

You do need a identity management solution that can use OAuth2/OIDC flows. This can be custom code or something like Ory Kratos - again what makes sense depends ALOT on your use case details and requirements. 
Ory offers support for this as well, [feel free to reach out](https://www.ory.sh/contact/). Also all of this is integrated and working out of the box on the managed [Ory Network](https://www.ory.sh/network/).
